linera-witty/src/type_traits/implementations/std/vec.rs

```diff
@@ -14,3 +17,40 @@ where
 
     type Layout = HList![i32, i32];
 }
+
+impl<T> WitLoad for Vec<T>
+where
+    T: WitLoad,
+{
+    fn load<Instance>(
+        memory: &Memory<'_, Instance>,
+        location: GuestPointer,
+    ) -> Result<Self, RuntimeError>
+    where
+        Instance: InstanceWithMemory,
+        <Instance::Runtime as Runtime>::Memory: RuntimeMemory<Instance>,
+    {
+        let address = GuestPointer::load(memory, location)?;
+        let length = u32::load(memory, location.after::<GuestPointer>())?;
```

Comment:
What's the tradeoff between computing offsets at compile-time (here) vs on the flight at run-time (serde)?
Do we need a random access to parts of the data in the future?

Reply:
Let's talk about it offline